src/AccessibilityInsights.SharedUx/Controls/HierarchyControl.xaml.cs

```diff
@@ -644,13 +643,13 @@ private void RequestRefreshBasedOnTreeViewChange(TreeViewMode mode)
         {
             Configuration.TreeViewMode = mode;
             SetTreeViewModeOnSelectAction(mode);
+            SetFocusOnHierarchyTree();
 
             if (this.SelectedElement != null)
             {
                 // refresh tree automatically.
                 this.HierarchyActions.RefreshHierarchy(true);
             }
-            SetFocusOnHierarchyTree();
```

Comment:
This was actually a NOP, since the focus had been shifted on menu load. The focus shift needs to occur before refreshing the hierarchy, hence setting the focus earlier in this method.